contracts/zero-ex/test/features/fund_recovery_tests.ts

```diff
@@ -0,0 +1,126 @@
+            
+        });
+        blockchainTests.resets('Should delegatecall `transferTrappedTokensTo` from the exchange proxy', () => {
+            const ETH_TOKEN_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
+            const ZERO_EX_EXCHANGE_PROXY = '0xdef1c0ded9bec7f1a1670819833240f027b25eff';
+            it('Tranfers an arbitrary ERC-20 Token', async () => {
+                const amountOut = Web3Wrapper.toBaseUnitAmount(new BigNumber(100), 18);
+                const designatedAddressBalance = await token.balanceOf(designatedAddress).callAsync();
+                const epBalance = await token.balanceOf(zeroEx.address).callAsync();
+                await zeroEx.transferTrappedTokensTo(token.address, amountOut, designatedAddress).awaitTransactionSuccessAsync({ from: owner });
+                const epBalanceNew = await token.balanceOf(zeroEx.address).callAsync();
+                const designatedAddressBalanceAferTransfer = await token.balanceOf(designatedAddress).callAsync();
+                expect(designatedAddressBalanceAferTransfer.c![0]).to.equal(amountOut.c![0]);
+            });
+            it('Amount -1 transfers entire balance', async () => {                
+                const designatedAddressBalance = await token.balanceOf(designatedAddress).callAsync();
+                const balanceOwner = await token.balanceOf(zeroEx.address).callAsync();
+                const tx = await zeroEx.transferTrappedTokensTo(token.address, constants.MAX_UINT256 , designatedAddress).awaitTransactionSuccessAsync({ from: owner });
+                const balanceNew = await token.balanceOf(zeroEx.address).callAsync();
+                const designatedAddressBalanceAferTransfer = await token.balanceOf(designatedAddress).callAsync();
+                expect(balanceNew.c![0]).to.equal(0);
+                expect(designatedAddressBalanceAferTransfer.c![0]).to.equal(balanceOwner.c![0]);                
+            });
+            it('Transfers ETH ', async () => {
+                //connect to mainnet contract address
+                const zrxContractMainnet = new IZeroExContract(ZERO_EX_EXCHANGE_PROXY, env.provider, env.txDefaults);
+                //await weth.withdraw(wethBal);
+                const bal = await env.web3Wrapper.getBalanceInWeiAsync(zrxContractMainnet.address);
+                //const designatedAddressBalance = await token.balanceOf(designatedAddress).callAsync(); 
+                const tx = await zrxContractMainnet.transferTrappedTokensTo(ETH_TOKEN_ADDRESS, constants.MAX_UINT256 , designatedAddress).awaitTransactionSuccessAsync({ from: owner });
+                //const bal2 = await env.web3Wrapper.getBalanceInWeiAsync(zrxContractMainnet.address);
+                const designatedAddressBalance = await env.web3Wrapper.getBalanceInWeiAsync(zrxContractMainnet.address);
+               
+                
+            });
+            it('Feature `transferTrappedTokensTo` can only be called by owner', async () => {
```

Comment:
Check out [this](https://github.com/0xProject/protocol/blob/development/contracts/zero-ex/test/features/ownable_test.ts#L61-L66) for reference